### PR TITLE
The opencv build, when the env var CARGO_FEATURE_GEN_FEATS is set, wi…

### PR DESCRIPTION
**Pull request by ereichert**
…ll update the features listed in the Cargo.toml file. It derives the list of features from the CMakeLists.txt file in the downloaded source code. This commit also includes changes to minimize required work when building and generating features.

### CODE REVIEW

**Comment by ereichert** [comment]
@ryandbair I think this is done.

The command line I used for testing is:

`RUST_BACKTRACE=1 CARGO_FEATURE_BUILD=1 CARGO_FEATURE_WITH_FFMPEG_OFF=1 CARGO_FEATURE_WITH_TIFF_OFF=1 CARGO_FEATURE_WITH_PNG_ON=1  CARGO_FEATURE_BUILD_SHARED_LIBS_OFF=1 cbd -vv`

**Comment by ryandbair** [comment]
I'm not able to get the PNGs to build, both with the command above as well as compiling with tlx2 and the with_png_on flag set.

File: Cargo.toml [hunk 1]
```diff
@@ -32,13 +32,306 @@ path = "examples/video_to_gray.rs"
...
+
+# DO NOT MOVE THE FEATURES SECTION.
+# The features section is genertated by the build.rs script.
+# build.rs relies on the features section being at the end of the file.
+# The features section and content below will be completely replaced.
+# Any manual edits should be above the features section.
+[features]
+build = []
+with_1394_on = ["build"] #default = ON   IF (NOT ANDROID AND NOT IOS)
+with_1394_off = ["build"] #default = ON   IF (NOT ANDROID AND NOT IOS)
```
Comment (ryandbair) [inline Cargo.toml]: Wouldn't this make both the on and off flags set if build is set?

**Comment by ryandbair** [comment]
I should have used build_png_on, that works fine.

**Comment by ereichert** [comment]
I used with_png_on and that's specified in the TLX 2 build now as well. But looking at it further, I'm now sure if with_png_on should work on its own.

Thing is I've built TLX 2 multiple times without an issue. So I'm not really sure what's going on with that.

**Comment by ereichert** [comment]
> Wouldn't this make both the on and off flags set if build is set?

That comment isn't showing up on the PR, but featureA = ["build"] means that featureA depends on the build feature. I think that's so you can specify featureA in a consumer and not have to specify build.

**Comment by ryandbair** [comment]
I'm getting this failure on TLX2 (it builds fine by itself):

`
thread 'main' panicked at 'Failed to build opencv via build script.: "Failed to copy /home/rbair/src/trafficland/tlx2-review/target/debug/build/opencv-4ad5150afc6c3f37/out/dist/share/OpenCV/3rdparty/lib/liblibpng.a to /home/rbair/src/trafficland/tlx2-review/target/debug/build/opencv-4ad5150afc6c3f37/out/dist/share/OpenCV/3rdparty/lib/libpng.a."', ../src/libcore/result.rs:837
`

The file actually exists at: `target/debug/build/opencv-4ad5150afc6c3f37/out/build/3rdparty/lib/liblibpng.a`, not in dist. I'm not sure where the discrepancy is coming from.

**Comment by ereichert** [comment]
try adding build_png_on and with_png_on to the TLX 2 features. And do a clean.

**Comment by ryandbair** [comment]
Same deal with both features enabled. 

liblibpng.a exists, so I know its build built at least. Either the build isn't moving it to the right place or the script is expecting it to be in the wrong place. When you build, where does liblibpng.a end up?

**Comment by ereichert** [comment]
I cargo cleaned both TLX 2 and opencv. Built opencv-rust using the command above and TLX 2 with the features specified in the PR for TLX 2.

No problems. 

So there's something different about our configs that's causing an issue.

**Comment by ryandbair** [comment]
build_shared_libs_off and build_png_on was required to build on tlx2

**Pull request merged**